.gitignore

```diff
@@ -0,0 +1,7 @@
+./node_modules
+./dist
+./src/logs/*
+./src/temp/*
+./src/svg/*
+./src/configs/production.json
```

Comment:
I don't think we have any configs now, but I would create a `src/utils` folder where we could have all kinds of things like configs, constants etc...

Reply:
This folder is suppose to be used in `v2` to have a separate configurations files, one for `production`, `testing` and `development`.

I'm thinking of keeping `utils` folder for the functions that will be created in the future, as i'm thinking to separate the common used utils and put then in the `utils` folder like, the svg customizer class, font customizer class, `serializer`, `svgToPng`, and `zipFolder`, etc.. to depend on them instead of the duplication in the controller. 

Helpers folder will also be created to contain the helpers like the errorHandler, respond, and logger helpers. 

for that reason I decided to go with the separate configs folder, to contain the configurations json files, what do you think?

Reply:
I mean, probably I'm being picky here ;P
We can go with the folder structure you suggested: utils, helpers, and config.

For the config part, do we need a lot of configuration? 
I was thinking that maybe we want to handle thins in a single `src/config.json` or `.env`  if there are just a few of them (CORS and PORT).

But again, I'm just being picky, whatever you decide will work =)

Reply:
it might not be logical to have all of this separation in this milestone, but I think later we will get benefit from this separation when things go larger, I think. 

For the config part, I can see that we will not have that much of configurations (for now) probably, for `v2` it will be some configurations related to Redis, MongoDB, and algolia, I think that's it, in addition to PORT, CORS configs. 
The idea is, I couldn't find a way, to have separate configs for different environments using https://www.npmjs.com/package/dotenv for that reason I thought of using https://www.npmjs.com/package/config to have a separate configuration file for each environment ( `test`, `production`, `dev` ) where settings might differ. 

I'm just sharing my thoughts, if you see that it's unnecessary overhead, we can go with single .env file

Reply:
If we do future-prof, we can do the folder structure. None is an issue =)

Reply:
The only caveat of using single config file is that, we will need sometimes to change the env and services connection links (algolia, mongo, or other third party services like the translation api that will be used), manually, when we're testing something or developing something locally. if this's not an issue, and i think it's not, it make sense to go with single file. 

What's your decision?

Reply:
I will go with what you've suggested, using single config file. If we faced a really need for the separation, we can separate it then.

Reply:
I've updated the configurations flow, we will use a separate default.json file, and I've added an `index.ts` file to map the configs for typescript to infer the types correctly.

Reply:
It looks great, thank you! 
I think we're ready to merge, right? =D

Reply:
Thank you for following up with this, Yes we're, I will re-write the class right now

Reply:
Changes are committed